frontend/lib/framework/error-classes-list.cpp

```diff
@@ -128,6 +129,68 @@ void ErrorMissingInclude::write(ErrorWriterBase& wr) const {
   wr.note(moduleInclude, "expected file at path '", filePath, "'");
 }
 
+void ErrorUseImportUnknownSym::write(ErrorWriterBase& wr) const {
+  auto visibilityClause = std::get<const uast::VisibilityClause*>(info);
+  auto symbolName = std::get<std::string>(info);
+  auto searchedScope = std::get<const resolution::Scope*>(info);
+  auto useOrImport = std::get<const resolution::VisibilityStmtKind>(info);
+  wr.heading(kind_, type_, locationOnly(visibilityClause),
+             "cannot find symbol '", symbolName, "' for ", useOrImport, ".");
+  wr.message("Symbol named here:");
+  wr.code(visibilityClause);
+  wr.message("Searching in scope of '", searchedScope->name(), "':");
+  wr.code(searchedScope->id());
+}
+
+void ErrorUseImportUnknownMod::write(ErrorWriterBase& wr) const {
+  auto id = std::get<const ID>(info);
+  auto moduleName = std::get<std::string>(info);
+  auto useOrImport = std::get<const resolution::VisibilityStmtKind>(info);
+  wr.heading(kind_, type_, id, "cannot find module '", moduleName,
+             "' for ", useOrImport, ".");
+  wr.message("Module named here:");
+  wr.code(id);
+}
+
+void ErrorUseImportNotModule::write(ErrorWriterBase& wr) const {
+  auto id = std::get<const ID>(info);
+  auto moduleName = std::get<std::string>(info);
+  auto useOrImport = std::get<const resolution::VisibilityStmtKind>(info);
+  wr.heading(kind_, type_, id, "attempted to ", useOrImport, " '", moduleName,
+             "' which is not a module.");
+  wr.message("Target named here:");
+  wr.code(id);
+}
+
+void ErrorAsWithUseExcept::write(ErrorWriterBase& wr) const {
+  auto use = std::get<const uast::Use*>(info);
+  auto as = std::get<const uast::As*>(info);
+  wr.heading(kind_, type_, locationOnly(use),
+             "invalid use of 'as' in this use statement.");
+  wr.note(locationOnly(as), "cannot use 'as' with 'except'.");
+  wr.message("Renaming in 'except'-lists is not supported.");
+}
+
+void ErrorDotExprInLimitations::write(ErrorWriterBase& wr) const {
+  auto dot = std::get<const uast::Dot*>(info);
+  wr.heading(kind_, type_, locationOnly(dot), "encountered dot expression '",
+             dot, "' in visibility limitation.");
+  wr.message(
+      "Dot expressions are not allowed in the limitations of a visibility "
+      "statement (use/import).");
```

Comment:
Uses and imports are visibility statements. I was concerned about a user not knowing that which is why I wrote `(use/import)`, but it might be better to simply write out uses and imports as you suggested.

As for 'limitation', it covers both `except` and `only`, but it is not a spec word (just a source code word) so I'll write out those keywords in the error message.